nginx/ngx_http_js_module.c

```diff
@@ -6184,9 +6202,18 @@ ngx_http_qjs_variables_own_property(JSContext *cx, JSPropertyDescriptor *pdesc,
         return 1;
     }
 
-    key = ngx_hash_strlow(name.data, name.data, name.len);
+    name_lc.data = (name.len < sizeof(storage))
```

Comment:
It would be great if we support VLA(variable length array), I mean do it like this.
```
storage  [val.length];
```
Then we can get rid of memory allocation.

Reply:
VLA are not very portable. 
Also, fixed buffer size helps us to ensure the more predictable memory consumption.